src/transformers/adapters/heads.py

```diff
@@ -483,7 +483,14 @@ def set_active_adapters(
             elif isinstance(final_block, Parallel):
                 self.active_head = [a if isinstance(a, str) else a.last for a in final_block.children]
             elif isinstance(final_block, BatchSplit):
-                self.active_head = final_block
+                blocks = [
+                    block.last() if isinstance(block, AdapterCompositionBlock) else block for block in final_block
```

Comment:
```suggestion
                    block.last if isinstance(block, AdapterCompositionBlock) else block for block in final_block
```

Reply:
This needs to be a method call otherwise it does not work

Reply:
Oh, sorry, thought it was a property based on `a.last` four lines above (which in that case also should not work).